pkg/patterns/declarative/metrics.go

```diff
@@ -345,21 +345,22 @@ func (gvkt *gvkTracker) deleteMetricsIfNeeded(metricsDuration int) {
 }
 
 func (gvkt *gvkTracker) start(ctx context.Context) error {
-	return gvkt.src.Start(ctx, gvkt.eventHandler, dummyQueue{}, gvkt.predicate)
+	return gvkt.src.Start(ctx, workqueue.NewTypedRateLimitingQueueWithConfig(
+		workqueue.DefaultTypedControllerRateLimiter[reconcile.Request](),
+		workqueue.TypedRateLimitingQueueConfig[reconcile.Request]{}))
 }
 
 func newGVKTracker(mgr manager.Manager, obj *unstructured.Unstructured, namespaced bool) (gvkt *gvkTracker) {
 	gvkt = &gvkTracker{}
 	gvkt.list = newItems()
 	gvkt.recorder = objectRecorderFor(obj.GroupVersionKind())
-	gvkt.src = commonclient.SourceKind(mgr.GetCache(), obj)
+	gvkt.src = commonclient.SourceKindWithHandler(mgr.GetCache(), obj, &recordTrigger{gvkt.list, namespaced, gvkt.recorder})
 	gvkt.predicate = predicate.Funcs{}
-	gvkt.eventHandler = commonclient.WrapEventHandler(recordTrigger{gvkt.list, namespaced, gvkt.recorder})
 
-	return
+	return gvkt
```

Comment:
Given we have a named return variable here, the return should be naked. If you prefer the return to include the variable, we should remove the name in the signature (which is also fine by me!):

```
func newGVKTracker(...) (*gvkTracker) {
    gvtk := &gvkTracker{}
     ...
```

Reply:
fixed!